### PR TITLE
Even more improved token validation script

### PR DESCRIPTION
**Pull request by randomshinichi**
Care has been taken to make the output of the script non-noisy, easy to read and know what to do next.
No unittests yet however.

Please review algorithms. I can break out validated-tokens.csv changes into a different PR.

### CODE REVIEW

File: src/logic.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import { ValidatedTokensData } from "./types/types";
+import { indexToLineNumber } from "./utils/validate";
+import { parse } from "csv-parse/sync";
+import fs from "fs";
+
+export async function validateValidatedTokensCsv(filename: string) {
+    const [records, recordsRaw] = parseCsv(filename);
+
+    const recordsPreviousRaw = await gitPreviousVersion("validated-tokens.csv");
+    fs.writeFileSync(".validated-tokens-0.csv", recordsPreviousRaw);
```
Comment (9yoi) [inline src/logic.ts:13]: if there are 300 open prs and the github action runs on each -- what is the previous raw?

File: src/utils/validate.ts [hunk 2]
```diff
@@ -1,58 +1,154 @@
...
+    }
+  });
 
-  const [tokenName, symbol, mint, decimals, imageURL, isCommunity] =
-    patch.added.lines[0].split(",");
+  const allowedDuplicateSymbols = [
+    'ALL', 'ARB', 'AVAX',
+    'BOO', 'FOOD', 'FUEL',
+    'GEAR', 'GM', 'LILY',
+    'MILK', 'NANA', 'NINJA',
```
Comment (9yoi) [inline src/utils/validate.ts:53]: what if someone pushes a third token that duplicates these?

File: src/logic.ts (hunk 1 above)
Comment (randomshinichi) [reply src/logic.ts:13]: it's always `main`... git takes care of that

File: src/utils/validate.ts (hunk 2 above)
Comment (randomshinichi) [reply src/utils/validate.ts:53]: yeah then it's gonna pass...

Comment (randomshinichi) [reply src/utils/validate.ts]: fixed with commit defaac4

**Review by 9yoi — APPROVED** [review]

**Pull request merged**